test/project_types/extension/models/specification_handlers/checkout_ui_extension_test.rb

```diff
@@ -218,6 +218,31 @@ def test_l10n_sum_max_filesize
           end
         end
 
+        def test_l10n_invalid_file_encoding
+          File.stubs(:read).returns("{\"invalid\": \"\xD8\x3D\xDC\xA9\" }")
+          write("locales/en.default.json", '{"laugh": "lol"}') # read content overriden by stub
```

Comment:
`write` fails with the return value of the stub?

Reply:
I had issues writing what would be considered invalid, the stub keeps things straightforward, but yeah, it's slightly ugly.